Testscripts/Windows/SETUP-DiffDiskGrowth.ps1

```diff
@@ -167,6 +161,12 @@ function Main {
         return $False
     }
 
+    $vmGeneration = Get-VMGeneration $vmName $hvServer
+    if (( $controllerType -eq "IDE" -or $vhdFormat -eq "vhd" ) -and $vmGeneration -eq 2 ) {
```

Comment:
have you debug this and get the value of $vhdFormat?

Reply:
yes, can get the value of $vhdFormat